### PR TITLE
Protocol identifiers are strings, listed in the API doc., and shouldn't normally be used

### LINKED ISSUE #1332: Usage of protocol identifiers
From the review by Lars Eggert, DISCUSS part:

### Section 6.1.3, paragraph 6
```
    In order to scope an alias to a specific transport protocol, an
    Endpoint can specify a protocol identifier.

    AlternateRemoteSpecifier.WithProtocol(QUIC)
```
This is the first and only time protocol identifiers are used. What
are they defined to be?

### LINKED ISSUE #1333: WithProtocol TCP/TLS in an example
From the review by Lars Eggert, DISCUSS part:

### Section 6.1.3, paragraph 9
```
    The following example shows a case where [example.com](http://example.com/) has a server
    running on port 443, with an alternate port of 8443 for QUIC.

    RemoteSpecifier := NewRemoteEndpoint()
    RemoteSpecifier.WithHostname("[example.com](http://example.com/)")
    RemoteSpecifier.WithPort(443)

    QUICRemoteSpecifier := NewRemoteEndpoint()
    QUICRemoteSpecifier.WithHostname("[example.com](http://example.com/)")
    QUICRemoteSpecifier.WithPort(8443)
    QUICRemoteSpecifier.WithProtocol(QUIC)

    RemoteSpecifier.AddAlias(QUICRemoteSpecifier)
```
Why does the `RemoteSpecifier` definition not contain a `WithProtocol`
clause for TCP/TLS? And what would that look like, given that TCP/TLS
is a protocol combination?

### LINKED ISSUE #1361: WithPort and WithService, but not WithProtocol
From the review by Paul Wouters, DISCUSS part:

How does one choose their syslog service transport protocol, since syslog
over udp and tcp are valid and both have the same service name? Or does
WithService accept values like "514/udp" ?

### PR DESCRIPTION
**Pull request by mwelzl**
Closes #1332
Closes #1333
Closes #1361

### CODE REVIEW

**Review by gorryfair — APPROVED** [review]

File: draft-ietf-taps-interface.md [hunk 1]
```diff
@@ -869,8 +869,18 @@ multiple aliases set.
...
+an Endpoint can specify a protocol specifier. Protocol specifiers
+are strings where slashes separate protocol names. The API
+documentation of a Transport Services system implementation ought to
+contain a list of selectable protocol specifiers.
+
+Normally, an endpoint ought not to request a specific transport protocol or
+protocol stack. The Transport Services system is responsible for mapping
+the API to a specific available transport protocol stack and managing
+the available network interfaces and paths. When specifically needed,
+this automated selection could be over-ridden (e.g., for testing or
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:882]:
```suggestion
this automated selection could be overridden (e.g., for testing or
```

File: draft-ietf-taps-interface.md [hunk 2]
```diff
@@ -869,8 +869,18 @@ multiple aliases set.
 RemoteSpecifier.AddAlias(AlternateRemoteSpecifier)
 ~~~
 
-To scope an alias to a specific transport protocol, an Endpoint can
-specify a protocol specifier.
+To scope an alias to a specific transport protocol or protocol stack,
+an Endpoint can specify a protocol specifier. Protocol specifiers
+are strings where slashes separate protocol names. The API
+documentation of a Transport Services system implementation ought to
+contain a list of selectable protocol specifiers.
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:876]:
```suggestion
To scope an alias to apply conditionally to a specific transport
protocol (such as defining an alternate port to use when QUIC
is selected, as opposed to TCP), an alias Endpoint can be
associated with a protocol identifier. Protocol identifiers are
objects or enumeration values provided by the Transport
Services API, which will vary based on which protocols are
implemented in a particular system.

specify a protocol specifier. Protocol specifiers
are strings where slashes separate protocol names. The API
documentation of a Transport Services system implementation ought to
contain a list of selectable protocol specifiers.
```

File: draft-ietf-taps-interface.md [hunk 3]
```diff
@@ -869,8 +869,18 @@ multiple aliases set.
...
+
+Normally, an endpoint ought not to request a specific transport protocol or
+protocol stack. The Transport Services system is responsible for mapping
+the API to a specific available transport protocol stack and managing
+the available network interfaces and paths. When specifically needed,
+this automated selection could be over-ridden (e.g., for testing or
+debugging purposes).
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:883]:
```suggestion
```

Suggest removing this paragraph, since this is specific to aliases.

**Review by tfpauly — CHANGES_REQUESTED** [review]

File: draft-ietf-taps-interface.md [hunk 4]
```diff
@@ -869,8 +869,18 @@ multiple aliases set.
...
+are strings where slashes separate protocol names. The API
+documentation of a Transport Services system implementation ought to
+contain a list of selectable protocol specifiers.
+
+Normally, an endpoint ought not to request a specific transport protocol or
+protocol stack. The Transport Services system is responsible for mapping
+the API to a specific available transport protocol stack and managing
+the available network interfaces and paths. When specifically needed,
+this automated selection could be overridden (e.g., for testing or
+debugging purposes).
```
Comment (tfpauly) [inline draft-ietf-taps-interface.md:883]:
```suggestion
```

**Pull request merged**